fairworkflows/fairworkflow.py

```diff
@@ -489,25 +473,8 @@ def publish_as_nanopub(self, use_test_server=False, **kwargs):
         for step in self:
             if step.is_modified or not step._is_published:
                 self._is_modified = True  # If one of the steps is modified the workflow is too.
-                old_uri = step.uri
-                var_names = [var.name for var in (step.inputs + step.outputs)]
-
-                step.publish_as_nanopub(use_test_server=use_test_server, **kwargs)
-                published_step_uri = step.uri
-
-                replace_in_rdf(self.rdf, oldvalue=rdflib.URIRef(old_uri),
-                               newvalue=rdflib.URIRef(published_step_uri))
-
-                # Similarly replace old URIs for variable name bindings
-                published_step_uri_defrag, _ = urldefrag(published_step_uri)
-                for var_name in var_names:
-                    old_var_uri = old_uri + '#' + var_name
-                    new_var_uri = published_step_uri_defrag + '#' + var_name
-                    replace_in_rdf(self.rdf, oldvalue=rdflib.URIRef(old_var_uri),
-                                  newvalue=rdflib.URIRef(new_var_uri))
+                raise RuntimeError(f'{step} was not published yet, please publish steps first')
```

Comment:
I'm slightly confused here - the docstring says ```First publish the steps, use the URIs of the published steps in the workflow. Then publish the workflow.``` but this now looks like it refuses to publish any of the steps, and instead insists the user has already done so. Would it not be better if it would just publish them by default when user tries to publish the whole workflow? Or have I misunderstood the proposed usage here?

Reply:
Remzi and Tobias requested this, we discussed it during the hacking session I think when you were in the parallel session. The user should always 'manually' publish the steps. That also makes the interface with jupyter extension simpler. The docstring is not reflecting this indeed, so I updated that!